draft-pauly-quic-datagram.md

```diff
@@ -121,7 +121,11 @@ max_datagram_frame_size transport parameter MUST terminate the connection with
 error PROTOCOL_VIOLATION. An endpoint that receives a DATAGRAM frame that is
 strictly larger than the value it sent in its max_datagram_frame_size
 transport parameter MUST terminate the connection with error
-PROTOCOL_VIOLATION.
+PROTOCOL_VIOLATION. Endpoints that wish to use DATAGRAM frames need to ensure
+they send a max_datagram_frame_size value sufficient to allow their peer to
+use them. It is RECOMMENDED to send the value 65536 in the
+max_datagram_frame_size transport parameter as that indicates to the peer that
+this endpoint will accept any DATAGRAM frame that fits inside a QUIC packet.
```

Comment:
Do we really need to have a max_datagram_frame_size then? UDP protocols being ported to QUIC would have no need.

Reply:
I think it’s more about indicating when you have some upstream MTU constraint that you want to use as a limit. It’s a useful parameter in that case.

Reply:
I think it’s more about indicating when you have some upstream MTU constraint that you want to use as a limit. It’s a useful parameter in that case.

Reply:
In this case perhaps a distinct error code is warranted (YAEC). It is very easy to assume datagrams are just set to max and when that fails you have no clue why the connection breaks down.

Reply:
Having a value on this TP can be useful for some applications, so we can leave it for now. Applications that ported their protocol directly from UDP sockets can use the value 65536.

I think we can leave it up to the application to define their own error code if they have an error condition. It's possible someone could write an application that only ever sends 4-byte DATAGRAMs and they should be able to set this to 4 and have everything work.

Reply:
The error was more for debugging - that won't work with arbitrary error codes decided by the peer. But it was just a suggestion - it suppose is not easy to inject formal error codes from extensions. Note that in general I favor PROTOCOL_VIOLOATION, but that is because many errors relate to malicious patterns and then less information is a good thing. This case is somewhat different in that respect.